source/03-components/table/table--sortable.es6.ts

```diff
@@ -0,0 +1,295 @@
+import Drupal from 'drupal';
+import once from 'once';
+
+const TABLE = `.c-table.is-sortable`;
+const SORTED = 'aria-sort';
+const ASCENDING = 'ascending';
+const DESCENDING = 'descending';
+const SORT_OVERRIDE = 'data-sort-value';
+const SORT_BUTTON_CLASS = `c-table__header-button`;
+const SORT_BUTTON = `.${SORT_BUTTON_CLASS}`;
+const SORTABLE_ATTRIBUTE = `data-sortable`;
+const SORTABLE_HEADER = `th[${SORTABLE_ATTRIBUTE}]`;
+const ANNOUNCEMENT_CLASS = `c-table__announcement-region`;
+const ANNOUNCEMENT_REGION = `.${ANNOUNCEMENT_CLASS}[aria-live="polite"]`;
+const SORT_ALL_COLUMNS = true;
+
+/**
+ * Gets the data-sort-value attribute value, if provided — otherwise, gets
+ * the innerText or textContent of the child element (HTMLTableCellElement)
+ * at the specified index of the given table row.
+ *
+ * @param {number} index
+ * @param {HTMLTableRowElement} tr
+ * @return {string | number | null}
```

Comment:
Is this best practice? With TS, this seems unnecessary as the types are already declared. If we still want to do this, the params are out of order: `tr` is second in the JSDoc but first in the function.